packages/esm-patient-orders-app/src/order-basket/order-basket.workspace.tsx

```diff
@@ -62,23 +64,29 @@ const OrderBasket: React.FC<DefaultWorkspaceProps> = ({
           abortController,
         );
         mutateEncounterUuid();
+        clearOrders();
+        mutateOrders().then();
+        closeWorkspaceWithSavedChanges();
+        showOrderSuccessToast(t, orders);
       } catch (e) {
-        setCreatingEncounterError(e);
+        setCreatingEncounterError(
+          e.responseBody.error.message ||
+            t('tryReopeningTheWorkspaceAgain', 'Please try launching the workspace again'),
+        );
         console.error(e);
-        return; // don't try to create the orders if we couldn't create the encounter
       }
-    }
-
-    const erroredItems = await postOrders(orderEncounterUuid, abortController);
-    clearOrders({ exceptThoseMatching: (item) => erroredItems.map((e) => e.display).includes(item.display) });
-    mutateOrders();
-    if (erroredItems.length == 0) {
-      closeWorkspaceWithSavedChanges();
-      showOrderSuccessToast(t, orders);
     } else {
-      showOrderFailureToast(t);
+      const erroredItems = await postOrders(orderEncounterUuid, abortController);
+      clearOrders({ exceptThoseMatching: (item) => erroredItems.map((e) => e.display).includes(item.display) });
+      mutateOrders().then();
```

Comment:
Is it preferable to await the promise instead?

```suggestion
      await mutateOrders()
```